src/utils/status.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { Tooltip, Icon } from 'antd'
+
+const isAutomaticallyUpgradeEngine = (volume, engineUpgradePerNodeLimit, defaultEngineImage) => {
+  if (engineUpgradePerNodeLimit && engineUpgradePerNodeLimit.value !== '0') {
+    if (defaultEngineImage) {
+      return volume.engineImage !== defaultEngineImage.image
+    }
+    return false
+  }
+  return false
+}
```

Comment:
To determine whether a volume is upgrading, we only need to check `volume. engineImage  != volume. currentImage`
We don't need to compare it to the `defaultEngineImage`

```
export function statusUpgradingEngine(volume) {
   if (volume && volume. engineImage && volume. currentImage) {
      return volume. engineImage  != volume. currentImage
   }
   return false
 }
```